src/passes/FuncCastEmulation.cpp

```diff
@@ -38,7 +38,7 @@ namespace wasm {
 // This should be enough for everybody. (As described above, we need this
 // to match when dynamically linking, and also dynamic linking is why we
 // can't just detect this automatically in the module we see.)
-static const int NUM_PARAMS = 16;
+static const int NUM_PARAMS = 61;
```

Comment:
(Wow, SciPy needs that many...? Couldn't it be filling in a struct and passing a pointer to it maybe..? Probably they have a good reason though.)

This has a downside for both code size and runtime speed. How about making it a parameter to the pass? See the `ExtractFunction` pass for an example of a tiny pass that has a parameter. And it's then easy to tell emcc to pass the parameter to wasm-opt when it runs, with the `BINARYEN_EXTRA_PASSES` flag.